pkgs/build-support/rust/build-rust-crate/test/default.nix

```diff
@@ -94,6 +94,44 @@ let
         ''
       );
 
+    assertOutputs = { name, crateArgs, expectedFiles, output? null }:
```

Comment:
I think we should start adding some "docstring"-like comments to these function (as we do in crate2nix). During the discussion of #83379 I was told that currently it isn't really obvious how they work. I intend to refactor the test runner once all the current PRs have been merged and things have settled down.

Reply:
I am all for docs! I added some (push still outstanding).